crates/subspace-node/src/commands/run.rs

```diff
@@ -128,6 +131,10 @@ pub async fn run(run_options: RunOptions) -> Result<(), Error> {
         None
     };
 
+    if maybe_domain_configuration.is_some() {
+        ensure_block_and_state_pruning_params(&mut subspace_configuration.base)
```

Comment:
Remind me of another issue, since the consensus block is not mapping to the domain block 1:1 and the challenge period is defined in the domain block, for the same challenge period there can be much more than 14400 consensus block (considering some consensus block doesn't contains bundle).

I don't have any better idea ATM but for the last resort, we can set the consensus pruning to `archive-canonical` or we can put a TODO for now and revisit later.

Reply:
I have thought on this yes. 
As an initial thought, we will push bundles until there are no unconfirmed execution receipts with extrinsics. We will skip bundle production once the all the ERs which has atleast extrinsic is confirmed.

So given our assumption on having one honest operator, they will be able to produce a fraud proof immediately.

But I do see potential case when honest operator is not fully synced. While this is not a pronounced problem, lets keep it as is for now and I;ll leave a todo for this